File: src/Day1/calorie-counting.ts
export const createElfList = (calorieLog: number[]):number[] => {
    let elfList: number[] = [];
    let currentCalories: number = 0;
  
    calorieLog.forEach((logEntry: number) => {
      if(logEntry > 0) {
        currentCalories += logEntry;
      } else {
        elfList.push(currentCalories);
        currentCalories = 0;
      }
    });

    console.log(`elfList: ${elfList}`);
    return elfList;
}

export const maxCalorieEntry = (elfList: number[]):number => {
  return elfList.sort((a, b) => b - a).slice(0,1)[0];
};

export const caculateTopThreeElves = (elfList: number[]): number => {
  if(elfList.length > 3){
    const topThreeElves: number[] = elfList.sort((a, b) => a - b).slice(-3);
    return topThreeElves.reduce((a, b) => a + b);
  }
  return elfList.reduce((a, b) => a + b);
}

File: Day1/CalorieCounting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using AdventOfCode.Common;

namespace AdventOfCode.Day1
{
    internal class CalorieCounting
    {
        private List<int> CreateElfList(List<string> calorieLog)
        {
            var elfList = new List<int>();
            var currentCalories = 0;

            foreach (var logEntry in calorieLog)
            {
                if (logEntry.Length > 0)
                {
                    currentCalories += Int32.Parse(logEntry);
                }
                else
                {
                    elfList.Add(currentCalories);
                    currentCalories = 0;
                }
            }
            return elfList;
        }


        internal void Play()
        {
            List<string> calorieLog = InputLoader.LoadData("day1.txt", "1");

            var elfList = CreateElfList(calorieLog);
            elfList.Sort();
            var firstEntry = elfList[^1];
            var lastThreeEntries = elfList.GetRange(elfList.Count - 3, 3).Sum();

            Console.WriteLine($"Day1: Max calorie entry: {firstEntry}, Top three elves: {lastThreeEntries}");
        }
    }
}